brainscore/benchmarks/lee2020.py

```diff
@@ -0,0 +1,208 @@
+    ssa = np.einsum('ij,ij->i', a_ma, a_ma)  # var A
+    ssb = np.einsum('ij,ij->i', b_mb, b_mb)  # var B
+    return np.einsum('ij,ij->i', a_ma, b_mb) / np.sqrt(ssa * ssb)  # cov/sqrt(varA*varB)
+
+
+def pairwise_distances(assembly):
+    '''
+    Convenience function creating a simple lookup table for pairwise distances
+    :param assembly: NeuroidAssembly
+    :return: square matrix where each entry is the distance between the neuroids at the corresponding indices
+    '''
+    locations = np.stack([assembly.neuroid.tissue_x.data, assembly.neuroid.tissue_y.data]).T
+
+    return squareform(pdist(locations, metric='euclidean'))
+
+
+def create_pairwise_neuroid_reliability_mat(neuroid_reliability):
+    '''
+    Convenience function creating a simple lookup table for combined reliabilities of neuroid pairs
+    :param neuroid_reliability: expects Score object where neuroid_reliability.raw holds [cross validation subset,
+        reliability per neuroid]
+    :return: square matrix where each entry_ij = sqrt(reliability_i * reliability_j)
+    '''
+    reliability_per_neuroid = np.mean(neuroid_reliability.raw.data, axis=0)
+    c_mat = np.zeros((reliability_per_neuroid.size, reliability_per_neuroid.size))
+    for i, ci in enumerate(reliability_per_neuroid):
+        for j, cj in enumerate(reliability_per_neuroid):
+            c_mat[i, j] = np.sqrt(ci * cj)
+
+    return c_mat
+
+
+def sample_array_locations(neuroid, num_sample_arrs, array_size_mm):
+    '''
+    Generator: Sample Utah array-like portions from artificial model tissue and generate masks
+    :param neuroid: NeuroidAssembly.neuroid
+    :param num_sample_arrs: number of simulated Utah arrays you want to sample
+    :param array_size_mm: Tuple, physical size of the arrays you want to sample in mm
+    :return: list of masks in neuroid dimension of assembly, usage: assembly[mask] -> neuroids within one array
+    '''
+    bound_max_x, bound_max_y = np.max([neuroid.tissue_x.data, neuroid.tissue_y.data], axis=1) - array_size_mm
+    rng = np.random.default_rng()
+    lower_corner = np.column_stack((rng.choice(neuroid.tissue_x.data[neuroid.tissue_x.data <= bound_max_x],
+                                               size=num_sample_arrs),
+                                    rng.choice(neuroid.tissue_y.data[neuroid.tissue_y.data <= bound_max_y],
+                                               size=num_sample_arrs)))
+    upper_corner = lower_corner + array_size_mm
+
+    # create index masks of neuroids within sample windows
+    for i in range(num_sample_arrs):
+        yield np.logical_and.reduce([neuroid.tissue_x.data <= upper_corner[i, 0],
+                                     neuroid.tissue_x.data >= lower_corner[i, 0],
+                                     neuroid.tissue_y.data <= upper_corner[i, 1],
+                                     neuroid.tissue_y.data >= lower_corner[i, 1]])
+
+
+def tissue_update(assembly):
+    # temporary function until assemblies updated

```

Comment:
what does the function do? What specific update needs to happen for this method to become obsolete?